app/database/models/user.py

```diff
@@ -69,11 +69,15 @@ def __init__(self, name, username, password, email, terms_and_conditions_checked
         self.terms_and_conditions_checked = terms_and_conditions_checked
 
         # saving hash instead of saving password in plain text
-        self.set_password(password)
+        if not social_login:
+            self.set_password(password)
 
         # default values
         self.is_admin = True if self.is_empty() else False  # first user is admin
-        self.is_email_verified = False
+        if social_login:
+            self.is_email_verified = True
```

Comment:
Why not verify here? Any special usecase?

Reply:
Yes. For normal login, the email is not verified and the user specifically has to do that from a link they receive.

Reply:
Got you, I was thinking something different, not applicable here though

Reply:
I have now done - `self.is_email_verified = social_login`
That removes unnecessary if statements.